File: PHP/PapaleguaTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("Papalegua.php");

class PapaleguaTest extends TestCase {

    public function testPapalegua() {
        $obj = new Papalegua();
        //Test 1
        $want = 9.71;
        $got = $obj->theFaster([9.81, 9.71]);
        $this->assertEquals($want, $got);
        //Test 2
        $want = 9.72;
        $got = $obj->theFaster([9.82, 9.72, 9.92]);
        $this->assertEquals($want, $got);
        //Test 3
        $want = 9.23;
        $got = $obj->theFaster([9.75, 9.55, 9.23]); 
        $this->assertEquals($want, $got);
        //Test 4
        $want = 9.10;
        $got = $obj->theFaster([9.82, 9.92, 9.1]);
        $this->assertEquals($want, $got);
    }
}

File: JAVA/PapaleguaTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;

import org.junit.Test;

public class PapaleguaTest {

	@Test
	public void testPapalegua() throws Exception {
		//Test 1
		double want = 9.71;
        ArrayList<Double> ops = new ArrayList<>();
        ops.add((double) 9.81);
        ops.add((double) 9.71);
		double got = Papalegua.theFaster( ops );
        assertEquals(want, got, 0.01);
		//Test 2
		want = 9.72;
        ops = new ArrayList<>();
        ops.add((double) 9.82);
        ops.add((double) 9.72);
        ops.add((double) 9.92);
        got = Papalegua.theFaster( ops );
        assertEquals(want, got, 0.01);
	}

}
